core-container/entry.sh

```diff
@@ -16,88 +16,487 @@
+#   - deployments/scale
 
-kubestellar start
+#   - cronjobs/status
+#   - csistoragecapacities/status
+#   - daemonsets/status
+#   - endpoints/status
+#   - endpointslices/status
+#   - horizontalpodautoscalers/status
+#   - jobs/status
+#   - networkpolicies/status
+#   - persistentvolumeclaims/status
+#   - poddisruptionbudgets/status
+#   - podtemplates/status
+#   - replicasets/scale
+#   - replicasets/status
+#   - replicationcontrollers/scale
+#   - replicationcontrollers/status
+#   - statefulsets/scale
+#   - statefulsets/status
+#   verbs:
+#   - get
+#   - list
+#   - watch
+#   - update
+#   - patch
+# EOF
 
-# Create secrets in Kuberntes cluster
-echo "< Create secrets >-----------------------------------------"
+#     kubectl apply -f - <<EOF
+# apiVersion: rbac.authorization.k8s.io/v1
+# kind: ClusterRoleBinding
+# metadata:
+#   name: compute:apiexport:kubernetes-extended:bind
+# roleRef:
+#   apiGroup: rbac.authorization.k8s.io
+#   kind: ClusterRole
+#   name: compute:apiexport:kubernetes-extended:bind
+# subjects:
+# - apiGroup: rbac.authorization.k8s.io
+#   kind: Group
+#   name: apis.kcp.io:binding:system:authenticated
+# ---
+# apiVersion: rbac.authorization.k8s.io/v1
+# kind: ClusterRoleBinding
+# metadata:
+#   name: compute:authenticated:apiexport:kubernetes:maximal-permission-policy-extended
+# roleRef:
+#   apiGroup: rbac.authorization.k8s.io
+#   kind: ClusterRole
+#   name: compute:apiexport:kubernetes:maximal-permission-policy-extended
+# subjects:
+# - apiGroup: rbac.authorization.k8s.io
+#   kind: Group
+#   name: apis.kcp.io:binding:system:authenticated
+# EOF
 
-# set KUBECONFIG to empty string so that `kubectl` falls back on its in-cluster technique for reading credentials from the filesystem.
-KUBECONFIG=
+#     echo "Finished augmenting root:compute for KubeStellar"
+# }
 
-echo "Ensure secret in the current namespace..."
-if kubectl delete secret kubestellar 2> /dev/null; then
-    echo " Deleted secret in the current namespace."
-fi
-if [ -n "${EXTERNAL_HOSTNAME}" ]; then
-    if kubectl create secret generic kubestellar --from-file="${PWD}/.kcp/admin.kubeconfig" --from-file="${PWD}/.kcp/external.kubeconfig" 2> /dev/null; then
-        echo " Created secret in the current namespace."
-    else
-        echo " Not running in a cluster!"
+
+# function ensure_espw() {
+#     echo "Ensuring ESPW..."
+
+#     kubectl ws root &> /dev/null
+
+#     if kubectl get Workspace "$ESPW_NAME" &> /dev/null; then
+#         echo "ESPW workspace already exists -- using it:"
+#         kubectl ws "$ESPW_NAME"
+#     else
+#         kubectl ws create "$ESPW_NAME" --enter
+#     fi
+
+#     kubectl apply -f "/home/kubestellar/config/exports"
+
+#     echo "Finished populating the ESPW with KubeStellar APIExports."
+# }
+
+
+function run_kcp() {
+    echo "--< Starting kcp >--"
+
+    echo Attempting to delete kubestellar secret...
+    (
+        KUBECONFIG=
+        ! kubectl delete secret kubestellar
```

Comment:
Why the `!` here?
This is in the context of `set -e`, right? That means this command will fail the script if the Secret is not present at the start of the command, right? Wouldn't it be better to make the script not fail regardless of the presence or absence of the Secret before this command?

Reply:
my understanding and testing suggest that the ! avoids the script to fail on error with set -e

Reply:
I would expect an invocation of `kubectl` before launching the server to fail!  Why is this command here at all?  What good do you think it does?